.changeset/little-rice-promise.md

```diff
@@ -0,0 +1,13 @@
+---
+"wrangler": minor
+---
+
+feat: add `wrangler versions secret put`, `wrangler versions secret bulk` and `wrangler versions secret list`
+
+`wrangler versions secret put` allows for you to add/update a secret even if the latest version is not fully deployed. A new version with this secret will be created, the existing secrets and config are copied from the latest version.
+
+`wrangler versions secret bulk` allows you to bulk add/update multiple secrets at once, this behaves the same as `secret put` and will only make one new version.
+
+`wrangler versions secret list` lists the secrets available to the currently deployed versions. `wrangler secret list` will list for the latest version.
```

Comment:
I think this behaviour should be inverted. If `wrangler versions secret put/bulk` amends secrets from the _latest_ version, then `wrangler versions secret list` should list the secrets from the latest version.

For users not using gradual rollouts, `wrangler secret list` and `wrangler versions secret list` will output the secrets from the same (deployed) version. Therefore it's not a breaking change for `wrangler secret list` to list the secrets from the deployed version.

Reply:
Yeah agreed but we would break `wrangler secret list` behaviour today if we did make it output deployed versions (since right now it just dumps a JSON and we'd need to at least dump two JSONs + a header of what version it is)

Could do a `--latest` on `versions secret list for now`, these commands are still experimental and may change defaults later